pkg/netpol/connlist/connlist.go

```diff
@@ -384,21 +416,60 @@ func (ca *ConnlistAnalyzer) getConnectionsList(pe *eval.PolicyEngine) ([]Peer2Pe
 			}
 			allowedConnections, err := pe.AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer)
 			if err != nil {
-				ca.errors = append(ca.errors, newResourceEvaluationError(err))
 				return nil, err
 			}
 			// skip empty connections
 			if allowedConnections.IsEmpty() {
 				continue
 			}
-			connectionObj := &connection{
+			p2pConnection := &connection{
 				src:               srcPeer,
 				dst:               dstPeer,
 				allConnections:    allowedConnections.AllConnections(),
 				protocolsAndPorts: allowedConnections.ProtocolsAndPortsMap(),
 			}
-			res = append(res, connectionObj)
+			res = append(res, p2pConnection)
+		}
+	}
+	return res, nil
+}
+
+// getIngressAllowedConnections returns connections list from IngressAnalyzer intersected with PolicyEngine's connections
+func (ca *ConnlistAnalyzer) getIngressAllowedConnections(ia *ingressanalyzer.IngressAnalyzer,
+	pe *eval.PolicyEngine, peerList []eval.Peer) ([]Peer2PeerConnection, error) {
+	res := make([]Peer2PeerConnection, 0)
+	ingressConns, err := ia.AllowedIngressConnections()
+	if err != nil {
+		return nil, err
+	}
+	// adding the ingress controller pod to the policy engine,
+	ingressControllerPod, err := pe.AddPodByNameAndNamespace(ingressanalyzer.IngressPodName, ingressanalyzer.IngressPodNamespace)
+	if err != nil {
+		return nil, err
+	}
+	for _, peer := range peerList {
```

Comment:
no need to iterate the peers list if the map `ingressConns` is empty (for example no Ingress or Route resources were configured)

Reply:
it might be more efficient to iterate only the peers from `ingressConns` map. since the keys there are the string of the peer and not the peer object itself, you can modify the returned map `ingressConns` to be from peer string to a new type (struct) that holds both ConnectionSet and the Peer object itself.

Reply:
changed the ia.IsEmpty to check the ingress and routes map (this func will not be called if the ingress-analyzer is empty).
updated the loop to iterate only on the ingress connections